.github/workflows/release.yaml

```diff
@@ -0,0 +1,66 @@
+---
+name: Release
+
+run-name: release ${{ inputs.tag }}
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to be released'
+        required: true
+
+permissions:
+  packages: write
+  id-token: write
+  contents: read
+
+jobs:
+  check:
+    name: preliminary-checks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Ensure version is properly set
+        run: |
+          test "$(yarn -s run get-version)" = "${{ inputs.tag }}"
+      - name: Give user instruction
+        if: failure()
+        run: echo "::error file=package.json,line=6::The tag ${{ inputs.tag }} must match the version"
+  publish:
+    runs-on: ubuntu-latest
+    needs: check
```

Comment:
any reason to run in a different job ? It might take more time to setup another worker

Reply:
Before we had a reason, but now, not anymore so good catch.